### PR TITLE
[decompiler] handle types without inspects better in all-types

### PR DESCRIPTION
**Pull request by water111**
This moves some types to the right files, and gives them parents/flag-asserts.

There weren't as many as I thought, which is nice.

### CODE REVIEW

File: decompiler/config/jak2/all-types.gc [hunk 1]
```diff
@@ -16076,6 +16231,31 @@
   )
 |#
 
+;; (deftype entity (res-lump)
+;;   ()
+;;   :flag-assert #x1b00000034
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:16237]: this is already defined elsewhere
```lisp
(deftype entity (res-lump)
  ((pad uint32 5))
  :method-count-assert 27
  :size-assert         #x34
  :flag-assert         #x1b00000034
  ;; Failed to read fields.
  (:methods
    (entity-method-22 () none 22) ;; (birth! (_type_) _type_ 22)
    (entity-method-23 () none 23) ;; (kill! (_type_) _type_ 23)
    (entity-method-24 () none 24) ;; (add-to-level! (_type_ level-group level actor-id) none 24)
    (entity-method-25 () none 25) ;; (remove-from-level! (_type_ level-group) _type_ 25)
    (entity-method-26 () none 26) ;; (get-level (_type_) level 26)
    )
  )
```

File: decompiler/config/jak2/all-types.gc [hunk 2]
```diff
@@ -16076,6 +16231,31 @@
...
+
+;; (deftype entity-race-mesh (entity)
+;;   ()
+;;   :flag-assert #x1d00000038
+;;   )
+
+;; (deftype entity-actor (entity)
+;;   ()
+;;   :flag-assert #x2100000050
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:16257]:
```lisp
#|
(deftype entity-actor (entity)
  ()
  :method-count-assert 33
  :size-assert         #x50
  :flag-assert         #x2100000050
  ;; Failed to read fields.
  (:methods
    (entity-actor-method-27 () none 27) ;; (next-actor (_type_) entity-actor 27)
    (entity-actor-method-28 () none 28) ;; (prev-actor (_type_) entity-actor 28)
    (entity-actor-method-29 () none 29) ;; (debug-print (_type_ symbol type) none 29)
    (entity-actor-method-30 () none 30) ;; (dummy-30 (_type_ entity-perm-status symbol) none 30)
    (entity-actor-method-31 () none 31)
    (entity-actor-method-32 () none 32)
    )
  )
|#
```

File: decompiler/config/jak2/all-types.gc [hunk 3]
```diff
@@ -16076,6 +16231,31 @@
...
+
+;; (deftype entity-camera (entity)
+;;   ()
+;;   :flag-assert #x1b00000050
+;;   )
+
+;; (deftype entity-nav-mesh (entity)
+;;   ()
+;;   :flag-assert #x1d00000038
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:16247]: defined already
```lisp
#|
(deftype entity-nav-mesh (entity)
  ()
  :method-count-assert 29
  :size-assert         #x38
  :flag-assert         #x1d00000038
  ;; Failed to read fields.
  (:methods
    (entity-nav-mesh-method-27 () none 27)
    (entity-nav-mesh-method-28 () none 28)
    )
  )
|#
```

File: decompiler/config/jak2/all-types.gc [hunk 4]
```diff
@@ -15637,6 +15762,36 @@
   )
 |#
 
+;; (deftype drawable-inline-array-tfrag (drawable-inline-array)
+;;   ()
+;;   :flag-assert #x1100000064
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:15768]: was already defined with more information:
```lisp
#|
(deftype drawable-inline-array-tfrag (drawable-inline-array)
  ()
  :method-count-assert 17
  :size-assert         #x64
  :flag-assert         #x1100000064
  ;; Failed to read fields.
  (:methods
    )
  )
|#
```

File: decompiler/config/jak2/all-types.gc [hunk 5]
```diff
@@ -15310,6 +15430,11 @@
   )
 |#
 
+;; (deftype prototype-tie (drawable-inline-array)
+;;   ()
+;;   :flag-assert #x1100000064
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:15436]:
```lisp
#|
(deftype prototype-tie (drawable-inline-array)
  ()
  :method-count-assert 17
  :size-assert         #x64
  :flag-assert         #x1100000064
  ;; Failed to read fields.
  (:methods
    )
  )
|#
```

File: decompiler/config/jak2/all-types.gc [hunk 6]
```diff
@@ -15298,6 +15413,11 @@
   )
 |#
 
+;; (deftype drawable-inline-array-instance-tie (drawable-inline-array)
+;;   ()
+;;   :flag-assert #x1100000064
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:15419]:
```lisp
#|
(deftype drawable-inline-array-instance-tie (drawable-inline-array)
  ()
  :method-count-assert 17
  :size-assert         #x64
  :flag-assert         #x1100000064
  ;; Failed to read fields.
  (:methods
    )
  )
|#
```

File: decompiler/config/jak2/all-types.gc [hunk 7]
```diff
@@ -15121,6 +15221,21 @@
   )
 |#
 
+;; (deftype prototype-shrubbery (drawable-inline-array)
+;;   ()
+;;   :flag-assert #x1100000044
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:15227]:
```lisp
#|
(deftype prototype-shrubbery (drawable-inline-array)
  ()
  :method-count-assert 17
  :size-assert         #x44
  :flag-assert         #x1100000044
  ;; Failed to read fields.
  (:methods
    )
  )
|#
```

File: decompiler/config/jak2/all-types.gc [hunk 8]
```diff
@@ -10940,6 +11010,11 @@
   )
 |#
 
+;; (deftype gui-control (basic)
+;;   ()
+;;   :flag-assert #x1900000cd0
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:11016]:
```lisp
#|
(deftype gui-control (basic)
  ()
  :method-count-assert 25
  :size-assert         #xcd0
  :flag-assert         #x1900000cd0
  ;; Failed to read fields.
  (:methods
    (gui-control-method-9 () none 9)
    (gui-control-method-10 () none 10)
    (gui-control-method-11 () none 11)
    (gui-control-method-12 () none 12)
    (gui-control-method-13 () none 13)
    (gui-control-method-14 () none 14)
    (gui-control-method-15 () none 15)
    (gui-control-method-16 () none 16)
    (gui-control-method-17 () none 17)
    (gui-control-method-18 () none 18)
    (gui-control-method-19 () none 19)
    (gui-control-method-20 () none 20)
    (gui-control-method-21 () none 21)
    (gui-control-method-22 () none 22)
    (gui-control-method-23 () none 23)
    (gui-control-method-24 () none 24)
    )
  )
|#
```

File: decompiler/config/jak2/all-types.gc [hunk 9]
```diff
@@ -9273,6 +9313,21 @@
   )
 |#
 
+;; (deftype art-group (art)
+;;   ()
+;;   :flag-assert #xf00000020
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:9319]:
```lisp
#|
(deftype art-group (art)
  ()
  :method-count-assert 15
  :size-assert         #x20
  :flag-assert         #xf00000020
  ;; Failed to read fields.
  (:methods
    (art-group-method-13 () none 13) ;; (link-art! (_type_) art-group 13)
    (art-group-method-14 () none 14) ;; (unlink-art! (_type_) int 14)
    )
  )
|#
```

File: decompiler/config/jak2/all-types.gc [hunk 10]
```diff
@@ -7471,6 +7471,11 @@
   )
 |#
 
+;; (deftype drawable-inline-array-node (drawable-inline-array)
+;;   ()
+;;   :flag-assert #x1100000044
+;;   )
```
Comment (xTVaser) [inline decompiler/config/jak2/all-types.gc:7477]:
```lisp
#|
(deftype drawable-inline-array-node (drawable-inline-array)
  ()
  :method-count-assert 17
  :size-assert         #x44
  :flag-assert         #x1100000044
  ;; Failed to read fields.
  (:methods
    )
  )
|#
```

**Review by xTVaser — COMMENTED** [review]
noticed some of these new types are duplicates, they are already defined somewhere in all-types and with more information

File: decompiler/config/jak2/all-types.gc [hunk 11]
```diff
@@ -13798,6 +13878,11 @@
 ;; collide-target-h               ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+;; (deftype control-info (collide-shape-moving)
+;;   ()
+;;   :flag-assert #x4400001910
+;;   )
```
[the remark this answers is not in the document]
Comment (ManDude) [reply decompiler/config/jak2/all-types.gc:13884]: Yay!

[the remark this answers is not in the document]
Comment (ManDude) [reply decompiler/config/jak2/all-types.gc:13884]: On a positive note though, it appears to be 12588 bytes shorter than Jak 1's.

**Pull request merged**